### DISCUSSION TITLE
Start a Playlist as a schedule without using switch

### DISCUSSION THREAD

**Discussion by 918483** [savonet/liquidsoap#2699] [User Support]
I would like to start a playlist like a schedule via telnet command and when the time comes, it goes back to the playlist that was playing today, I use the command below, but when it's time to finish the playlist it waits for her to finish a song to return to the schedule, so taking a while almost 4 minutes to return would you have any solution?

Would there be any way to start this scheduling type via telnet being:

```
schedule.start <url>
```
and 
```
schedule.stop
```
how i use it today:

```
def dynplaylist_protocol(~rlog,~maxtime,arg)=
  print("Iniciando Sistema de Agendamento #{maxtime} - #{rlog} - #{arg}")
  ["/home/streaming/playlists/mute-mute.pls"]
end

add_protocol("dynplaylist", dynplaylist_protocol)
radio = playlist(id="agendamento", mode="normal", loop=false, "dynplaylist:dummy")
```

**Comment by 918483** [comment]
@toots  Would you help me ?

**Comment by 918483** [comment]
@smimram Would you help me ?

**Comment by rrolla** [comment]
@918483 looks like you need to pass `track_sensitive=false` you can read more on book on page 33 https://www.liquidsoap.info/book/book.pdf

**Comment by 918483** [comment]
I've already done that and managed to get to this script above, but when the schedule comes out it plays 2 more songs and then goes back to playing the main playlist if I skip 2 then it comes back I wanted to make a way that the time I sent it to start would be immediate and when ordering to leave it was immediate too sorry for my english

**Comment by toots** [comment]
@918483 have you looked at using a reference to a boolean?

```
should_play = ref(false)

s = switch(track_sensitive=false, [
  ({!should_play}, extra_playlist),
  ({true}, main_playlist)
])
```
Then you can register a telnet command to change `should_play` value. This should force the `extra_playlist` to be played or not.
